### DISCUSSION TITLE
mesuring resonances

### DISCUSSION THREAD

**Discussion by jlodew0** [xblax/flashforge_ad5m_klipper_mod#21] [Q&A]
@xblax how does the default FF procedure set the parameters for resonance compensation?
I remember that when, I connected an adxl345 to my rpi for my ender 3 pro, I mesured both axis seperate (not needed on core xy because the sensors doesn't need to be moved) and this generated 2 csv files that needed to be processed by ~/klipper/scripts/calibrate_shaper.py to get it in charts and then add the suggested frequencies in [input_shaper]

edit:
oh, I now see that there is an auto calibrate option as well: :sweat_smile:
https://www.klipper3d.org/Measuring_Resonances.html#input-shaper-auto-calibration

**Comment by xblax** [comment]
I have not yet tried if/how that works directly via Klipper. But I assume Flashforge is only using the built-in functions
